command/server.go

```diff
@@ -1225,6 +1237,37 @@ func (c *ServerCommand) AutocompleteFlags() complete.Flags {
 	}
 }
 
+// storePidFile is used to write out our PID to a file if necessary
+func (c *ServerCommand) storePidFile(pidPath string) error {
+	// Quit fast if no pidfile
+	if pidPath == "" {
+		return nil
+	}
+
+	// Open the PID file
+	pidFile, err := os.OpenFile(pidPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0666)
```

Comment:
These permissions are too open -- certainly the file should not be world-writable. I suggest 0644.